syntax = "proto3";

package glossary;

service GlossaryService {
  rpc GetTerms (Empty) returns (TermsList);
  rpc GetTerm (GetTermRequest) returns (Term);
  rpc CreateTerm (CreateTermRequest) returns (Term);
  rpc UpdateTerm (UpdateTermRequest) returns (Term);
  rpc DeleteTerm (DeleteTermRequest) returns (Empty);
}

message Term {
  string keyword = 1;
  string description = 2;
}

message TermsList {
  repeated Term terms = 1;
}

message GetTermRequest {
  string keyword = 1;
}

message CreateTermRequest {
  string keyword = 1;
  string description = 2;
}

message UpdateTermRequest {
  string keyword = 1;
  string description = 2;
}

message DeleteTermRequest {
  string keyword = 1;
}

message Empty {}
